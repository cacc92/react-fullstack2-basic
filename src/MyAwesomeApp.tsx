// export function MyAwesomeApp() {
//   return (
//     <>
//       <h1>César</h1>
//       <h3>Carrasco</h3>
//     </>
//   );
// }

import type { CSSProperties } from "react";

const firstName = "César";
const lastName = "Carrasco";
const favoriteGames = ["Call of Duty", "Battlefield", "Delta Force"];
const isActive = true;
const address = {
  zipCode: "ABC-123",
  country: "Chile",
};

const myStyle: CSSProperties = {
  backgroundColor: "#fafafa",
  padding: 10,
  borderRadius: 10,
  marginTop: 10,
};
// Tambien se puede resolver con una funcion tipo flecha
export const MyAwesomeApp = () => {
  return (
    <>
      <h1>{firstName}</h1>
      <h3>{lastName}</h3>

      <p>{favoriteGames.join(", ")}</p>

      <h1>{isActive ? "Esta activo" : "No esta activo"}</h1>

      <p style={myStyle}>{JSON.stringify(address)}</p>
    </>
  );
};
