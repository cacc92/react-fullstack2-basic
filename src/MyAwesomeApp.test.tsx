import { describe, expect, test } from "vitest";
import { MyAwesomeApp } from "./MyAwesomeApp";
import { render, screen } from "@testing-library/react";

describe("MyAwesomeApp", () => {
  test("should render firstname and lastname", () => {
    // ! No puedo generar el render directamente
    // const render  = <MyAwesomeApp />
    // console.log(render);

    // Como no puedo realizar el render porque no tengo un entorno de DOM
    // Para poder trabajar con JSX necesito un entorno de DOM
    // Ese entorno de DOM me lo da React Testing Library
    // npm install --save-dev @testing-library/react @testing-library/dom @types/react @types/react-dom

    // ! 1. Arrange (Preparacion)
    // console.log(document.body);
    const { container } = render(<MyAwesomeApp />);
    // screen.debug();
    // console.log(container.innerHTML);

    // ! 2. Act (Aplicar actiar)

    const h1 = container.querySelector("h1");
    const h3 = container.querySelector("h3");
    // console.log(h1?.innerHTML);

    // ! 3. Assert
    //expect(h1?.innerHTML).toBe("César");
    expect(h1?.innerHTML).toContain("César");
    expect(h3?.innerHTML).toBe("Carrasco");
  });

  test("should render firstname and lastname - screen", () => {
    // ! 1. Arrange (Preparacion)
    render(<MyAwesomeApp />);
    // screen.debug();

    // ! 2. Act (Aplicar actiar)

    // const h1 = screen.getByRole("heading", { level: 1 });
    const h1 = screen.getByTestId("first-name-title");
    console.log(h1.innerHTML);

    // ! 3. Assert
    expect(h1.innerHTML).toContain("César");
  });

  // Los snapshot son para verificar que la UI no ha cambiado
  // Si la UI cambia, el snapshot va a fallar
  // Si el cambio es intencional, se debe actualizar el snapshot
  // Cuando me refiero a UI, me refiero al HTML generado tiene que ser igual
  // Hablando de structura
  test("should match snapshot", () => {
    // ! 1. Arrange (Preparacion)
    const { container } = render(<MyAwesomeApp />);

    // ! 3. Assert
    expect(container).toMatchSnapshot();

    // screen.debug();
  });

  test("should match snapshot - screen", () => {
    // ! 1. Arrange (Preparacion)
    render(<MyAwesomeApp />);

    // ! 3. Assert
    expect(screen.getByTestId("div-app")).toMatchSnapshot();
  });
});
