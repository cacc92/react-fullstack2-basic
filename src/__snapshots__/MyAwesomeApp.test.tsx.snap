// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot - screen 1`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
     
    César
     
  </h1>
  <h3>
    Carrasco
  </h3>
  <p>
    2
  </p>
  <p>
    Call of Duty, Battlefield, Delta Force
  </p>
  <h1>
    Esta activo
  </h1>
  <p
    style="background-color: rgb(250, 250, 250); padding: 10px; border-radius: 10px; margin-top: 10px;"
  >
    {"zipCode":"ABC-123","country":"Chile"}
  </p>
</div>
`;

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-title"
    >
       
      César
       
    </h1>
    <h3>
      Carrasco
    </h3>
    <p>
      2
    </p>
    <p>
      Call of Duty, Battlefield, Delta Force
    </p>
    <h1>
      Esta activo
    </h1>
    <p
      style="background-color: rgb(250, 250, 250); padding: 10px; border-radius: 10px; margin-top: 10px;"
    >
      {"zipCode":"ABC-123","country":"Chile"}
    </p>
  </div>
</div>
`;
